src/AutoMapper/ConfigurationValidator.cs

```diff
@@ -11,10 +11,12 @@ namespace AutoMapper
     public class ConfigurationValidator
     {
         private readonly IConfigurationProvider _config;
+        private readonly Action<ValidationContext> _validator;
 
-        public ConfigurationValidator(IConfigurationProvider config)
+        public ConfigurationValidator(IConfigurationProvider config, Action<ValidationContext> validator)
```

Comment:
I like the idea here. But can we do this as part of IConfigurationProvider? I don't want to add more stuff to the constructor on this or the MapperConfiguration object